### PR TITLE
Dehexed water and pavement

### PR DESCRIPTION
**Pull request by SJuliez**
For the saxarba tileset, this makes water less hexy by making it use exits like roads do. This happens automatically and for existing maps. The behaviour can be overridden by manually setting the exits, so rivers exiting maps or other stuff can still be edited.

Also adds a version of saxarba's pavement that is also unhexed (pavement:4). It looks like standard pavement but does not have different light/dark versions for hex level (this is possible but would add another few hundred images). It works like the water does but does not get applied automatically to existing maps as it usually covers less area than the hex-y pavement and thus may not touch buildings or roads as intended.

image

image

### CODE REVIEW

**Comment by SJuliez** [comment]
I really don't know what the Validate Boards test does there

File: megamek/src/megamek/client/ui/swing/boardview/BoardView1.java [hunk 1]
```diff
@@ -2748,13 +2749,6 @@ private void drawHex(Coords c, Graphics boardGraph,
                 if (!useIsometric()) {
                     g.drawImage(scaledImage, 0, 0, this);
                 }
-                // draw a shadow for bridge hex.
-                if (useIsometric()
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/boardview/BoardView1.java:2752]: No more bridge shadows? :(

File: megamek/src/megamek/common/Hex.java [hunk 2]
```diff
@@ -181,6 +181,12 @@ public void setExits(IHex other, int direction, boolean roadsAutoExit) {
             }
 
             cTerr.setExit(direction, cTerr.exitsTo(oTerr));
+            
+            // Water gets a special treatment: Water at the board edge
+            // (hex == null) should usually look like ocean and 
+            // therefore always gets connection to outside the board 
+            if ((cTerr.getType() == Terrains.WATER) && (other == null))
+            	cTerr.setExit(direction, true);
```
Comment (NickAragua) [inline megamek/src/megamek/common/Hex.java:189]: one-liner if blocks should still get { }

**Review by NickAragua — COMMENTED** [review]
One question, one minor formatting quibble.

File: megamek/src/megamek/client/ui/swing/boardview/BoardView1.java (hunk 1 above)
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/boardview/BoardView1.java:2752]: They're still there! Just a few lines above that.

File: megamek/src/megamek/common/Hex.java (hunk 2 above)
Comment (SJuliez) [reply megamek/src/megamek/common/Hex.java:189]: That code bit is from 4 years ago when I hadnt seen the formatting guide :)

**Review by NickAragua — APPROVED** [review]

**Pull request merged**